pulsar/impl_message.go

```diff
@@ -21,36 +21,38 @@ import (
 	"errors"
 	"fmt"
 	"math"
-	"math/big"
-	"strings"
 	"sync"
 	"sync/atomic"
 	"time"
 
 	"google.golang.org/protobuf/proto"
 
 	pb "github.com/apache/pulsar-client-go/pulsar/internal/pulsar_proto"
+	"github.com/bits-and-blooms/bitset"
 )
 
 type messageID struct {
 	ledgerID     int64
 	entryID      int64
 	batchIdx     int32
 	partitionIdx int32
+	batchSize    int32
```

Comment:
#316 just changes the type from `int` to the portable `int32` that only occupies 4 bytes for all architectures. If you want to mention #324, it's also different. #324 moves some fields that are responsible for tracking the acknowledgment into the `MessageID` returned by consumer. For producer, only the required fields are reserved.

However, batch size field is a part of the Pulsar protocol. https://github.com/apache/pulsar-client-go/blob/44dc85c9323f5e64d1acf250fcafe8580617ce52/pulsar/internal/pulsar_proto/PulsarApi.proto#L64 Even the `MessageID` returned by producer should set this field.

>  otherwise the expression of messageID is not very correct,

Could you give an example?

Reply:
If you want to differ the non-batch message id with batch message id like Java's `MessageIdImpl` and `BatchMessageIdImpl`, the `batchIdx` and `batchSize` fields should be moved together.

```go
type messageID struct {
	ledgerID     int64
	entryID      int64
	partitionIdx int32
}

type batchMessageID struct {
        msgID messageID
	batchIdx     int32
	batchSize    int32
}
```

But I'm not sure if it could make the code complicated.